File: Civitas_Ruby/lib/sorpresa_pagarCobrar.rb
require_relative 'sorpresa.rb'
require_relative 'jugador.rb'
require_relative 'diario.rb'
module Civitas
  class Sorpresa_pagarCobrar < Sorpresa
    
    def initialize(valor, texto)
      super()
      @valor = valor
      @texto = texto
    end
    
    def aplicarAJugador(actual, todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].modificarSaldo(@valor)
      end
    end
    
    def informe(actual, todos)
      Diario.instance().ocurre_evento(todos[actual].to_s + " se ve afectado por una carta de tipo pagarCobrar y " + to_s())
    end
  end
end

File: Civitas_GUI/src/civitas/Sorpresa_pagarCobrar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;
/**
 *
 * @author francisco
 */
class Sorpresa_pagarCobrar extends Sorpresa {
    private int valor;
    
    Sorpresa_pagarCobrar(int valor1, String texto){
        super();
        this.valor = valor1;
        super.setTexto(texto);
    }
    
    @Override
    protected void informe(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual, todos))
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + " se ve afectado por una sorpresa de tipo Pagar/Cobrar " + super.toString());
    }
    
    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (actual > -1 && actual < todos.size()){
            this.informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
}
